src/qt/guiutil.cpp

```diff
@@ -818,10 +818,12 @@ void PolishProgressDialog(QProgressDialog* dialog)
     // Workaround for macOS-only Qt bug; see: QTBUG-65750, QTBUG-70357.
     const int margin = TextWidth(dialog->fontMetrics(), ("X"));
     dialog->resize(dialog->width() + 2 * margin, dialog->height());
-    dialog->show();
-#else
-    Q_UNUSED(dialog);
 #endif
+    // QProgressDialog estimates the time the operation will take (based on time
+    // for steps), and only shows itself if that estimate is beyond minimumDuration.
+    // The default minimumDuration value is 4 seconds, and it could make users
+    // think that the GUI is frozen.
+    dialog->setMinimumDuration(0);
```

Comment:
Why not just `show()`?

Reply:
From [QTBUG-47042](https://bugreports.qt.io/browse/QTBUG-47042?focusedCommentId=301657&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-301657):
> `QProgressDialog` is _designed_ to show itself automatically, based on an internal estimate for the duration of the operation and the `minimumDuration` property. You _never_ call `show()` or `exec()` on it manually. You're also not supposed to keep it around when it's not used. In 5.4, the only way to start the internal duration estimation was to call `setValue(0)`. But we noticed that many people didn't call `setValue(0)`, but just handed external progress information on to the progress dialog, and whether or not that progress information started at 0 or, say, 1, would determine whether the dialog would show itself at all, or not. This was thought to be a bit fragile, so we changed it so that the start time of the duration estimate was the dialog construction time unless `setValue(0)` (or, more correctly, `setValue(minimum())` was called, leading to the dialog showing itself `minimumDuration` after its construction, even if it was constructed not for immediate use (which the dialog cannot distinguish from a really long-running operation).